packages/iov-keycontrol/src/keyring.spec.ts

```diff
@@ -15,204 +15,226 @@ async function makeRandomEd25519Keypair(): Promise<Ed25519Keypair> {
+      expect(keyring.getWallets().length).toEqual(1);
+      expect(keyring.getWallets()[0]).toEqual(jasmine.any(Ed25519Wallet));
+    });
   });
 
-  it("can serialize one wallet", () => {
-    const keyring = new Keyring();
-    const wallet = Ed25519HdWallet.fromEntropy(fromHex("c7f74844892fd7b707e74fc9b6c8ef917c13ddbb380cadbc"));
-    keyring.add(wallet);
+  describe("add", () => {
+    it("can add one wallet", () => {
+      const keyring = new Keyring();
+      const wallet = Ed25519HdWallet.fromEntropy(
+        fromHex("065a823769888cbc84e8455522f0e1a066447cb1120aa92e6ee251b74257a7bf"),
+      );
 
-    expect(keyring.serialize()).toMatch(
-      /^{\"formatVersion\":1,\"wallets\":\[{\"implementationId\":\"ed25519-hd\",\"data\":\"{.*}\"}\]}$/,
-    );
-  });
+      keyring.add(wallet);
 
-  it("can serialize many wallets", () => {
-    const keyring = new Keyring();
-    const wallet1 = Ed25519HdWallet.fromEntropy(fromHex("c7f74844892fd7b707e74fc9b6c8ef917c13ddbb380cadbc"));
-    const wallet2 = Ed25519HdWallet.fromEntropy(fromHex("2a7e3f902279af82138f14f871badf8d92b33713eb6c7193"));
-    const wallet3 = Ed25519HdWallet.fromEntropy(fromHex("602c79484cf098bd4445ad45b5e6557d83ec743cebddb4cd"));
-    const wallet4 = Ed25519HdWallet.fromEntropy(fromHex("1124ef7ab681387eba8fdd93a0a88ec2f3326f2a6e5e967d"));
-    const wallet5 = Ed25519HdWallet.fromEntropy(fromHex("cd0e346ae5c714a1514562cb8ad5d4b5a2443dbbc5dd2b5b"));
-    const wallet6 = Ed25519HdWallet.fromEntropy(fromHex("e38dd5c066406668b51be00e8ad0276ed9dec967d95c2248"));
-    keyring.add(wallet1);
-    keyring.add(wallet2);
-    keyring.add(wallet3);
-    keyring.add(wallet4);
-    keyring.add(wallet5);
-    keyring.add(wallet6);
-
-    expect(keyring.serialize()).toMatch(
-      /^{\"formatVersion\":1,\"wallets\":\[{\"implementationId\":\"ed25519-hd\",\"data\":\"{.*}\"}(,{\"implementationId\":\"ed25519-hd\",\"data\":\"{.*}\"}){5}\]}$/,
-    );
-  });
+      expect(keyring.getWallets().length).toEqual(1);
+      // Ensure added wallet is the same object, no copy of it
+      expect(keyring.getWallets()[0]).toBe(wallet);
+    });
 
-  it("can deserialize empty", () => {
-    const keyring = new Keyring('{"formatVersion":1,"wallets":[]}' as KeyringSerializationString);
-    expect(keyring).toBeTruthy();
-    expect(keyring.getWallets().length).toEqual(0);
-  });
+    it("can add and get multiple wallets", () => {
+      const keyring = new Keyring();
+      const wallet1 = Ed25519HdWallet.fromEntropy(fromHex("f7e7f1bbb327113a46fd3fa1020413de"));
+      const wallet2 = Ed25519HdWallet.fromMnemonic(
+        "flip hunt behind absorb blush proof hurry embody quantum start pencil rapid",
+      );
+      // Wallet 3 and 4 have the same seed. This is stupid but not the Keyring's problem
+      const wallet3 = Ed25519HdWallet.fromEntropy(
+        fromHex("3275a0acb9f697875d829119e4eda0f799afe5e8fb0bc7199c75ae19df610949"),
```

Comment:
Putting this string in a variable would make it clear it's the same in both cases.

Reply:
Thanks, done in add96e16669edce374f503f5a60bbea8b86a0cb0